pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -972,6 +972,40 @@ const (
 func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDeployment, releaseImage string, cdLog log.FieldLogger) (*reconcile.Result, error) {
 	areImagesResolved := cd.Status.InstallerImage != nil && cd.Status.CLIImage != nil
 
+	// In minimal install mode, don't even run the image resolver pod -- just trust the InstallerImageOverride
+	if minimal, err := strconv.ParseBool(cd.Annotations[constants.MinimalInstallModeAnnotation]); err == nil && minimal {
+		changed1 := false
+		if !areImagesResolved {
```

Comment:
I recommend you add a unit test for clearing the existing `InstallImagesNotResolvedCondition` once the correct variables are set.

Reply:
Something other than [this](https://github.com/openshift/hive/pull/2157/files#diff-b9685aa0bf796d1db13613fe411ed287146777107405bcc935ebd1d45d64ea47R626-R632)?

Reply:
Oh, you mean if the variables are somehow set but the condition is not? Can do...

Reply:
...except that's not going to work with the code as written, is it?

Reply:
Done. Good call.